pyuvdata/uvdata/uvdata.py

```diff
@@ -1469,56 +1469,54 @@ def _check_for_unprojected(self):
 
         return blt_mask
 
-    def rename_phase_center(self, old_name=None, new_name=None, cat_id=None):
+    def rename_phase_center(self, catalog_identifier, new_name):
         """
         Rename a phase center/catalog entry within a multi phase center data set.
 
         Parameters
         ----------
-        old_name : str
-            Optional argument, phase center name for the target to be renamed. Note that
-            either `old_name` or `cat_id` must be specified.
+        catalog_identifier : str or int
+            Unique identifier of a phase center to be renamed. If supplied as a str,
+            will be matched against the phase center name. Otherwise if an int, assumed
+            to be the catalog ID number.
         new_name : str
             New name for the phase center.
-        cat_id : int or list of int
-            Optional argument, phase center IDs number for the target to be renamed.
-            Note that either `old_name` or `cat_id` must be specified.
 
 
         Raises
         ------
         ValueError
             If attempting to run the method on a non multi phase center data set, if
-            `old_name` is not found as a key in `phase_center_catalog`, if `new_name`
-            already exists as a key in `phase_center_catalog`, or if attempting to
-            name a source "unprojected" (which is reserved).
+            `catalog_identifier` is not found in `phase_center_catalog`, or if
+            attempting to name a source "unprojected" (which is reserved).
         TypeError
-            If `new_name` is not actually a string.
+            If `new_name` is not actually a string or if `catalog_identifier` is not a
+            string or an integer.
         """
         if not self.multi_phase_center:
             raise ValueError(
                 "Cannot rename a phase center if multi_phase_center != True."
             )
-        if (old_name is None) == (cat_id is None):
-            raise ValueError(
-                "Either old_name or cat_id must be supplied (but not both)."
-            )
 
-        if cat_id is None:
+        if not isinstance(catalog_identifier, (str, int)):
+            raise TypeError("catalog_identifier must be a string or an integer.")
```

Comment:
Got another nit picky question on this one. @kartographer based on the block at 1517 I get the feeling catalog_identifier could also be a list since it is cast as a list. Should we check for that in this? Or check it is an iterable of a float?

Reply:
done